project_task_code/models/project_task.py

```diff
@@ -28,3 +29,41 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        args = args or []
+        tasks = []
+        code = self._name_search_parse_code(name)
+        if code:
+            # We are searching for codes
+            tasks = self.search(
+                [('code', '=ilike', code + '%')] + args,
```

Comment:
No need to add `'%'`. This is already implicit in the ilike operator. Unlike in plain SQL, where you have to add it explicitly (but with more control, you can put it at the end, at the beginning, both, or basically anywhere in the comparison string).